route/build/src/standalone/model_setup.f90

```diff
@@ -256,32 +283,111 @@ SUBROUTINE inFile_pop(ierr, message)  ! output
     case default
       ierr=20; message=trim(message)//'<time_units>= '//trim(t_unit)//': <time_units> must be seconds, minutes, hours or days.'; return
    end select
-   infileinfo_data(iFile)%convTime2Days = convTime2Days
+   inputfileinfo(iFile)%convTime2Days = convTime2Days
 
    ! get the reference julian day from the nc file
-   call process_time(trim(infileinfo_data(iFile)%unit),infileinfo_data(iFile)%calendar,infileinfo_data(iFile)%ncrefjulday,ierr, cmessage)
+   call process_time(trim(inputfileinfo(iFile)%unit),inputfileinfo(iFile)%calendar,inputfileinfo(iFile)%ncrefjulday,ierr, cmessage)
    if(ierr/=0) then; message=trim(message)//trim(cmessage)//' [ncrefjulday]'; return; endif
 
    ! populated the index of the iTimebound for each nc file
    if (iFile==1) then
-    infileinfo_data(iFile)%iTimebound(1) = 1
-    infileinfo_data(iFile)%iTimebound(2) = nTime
+    inputfileinfo(iFile)%iTimebound(1) = 1
+    inputfileinfo(iFile)%iTimebound(2) = nTime
    else ! if multiple files specfied in the txt file
-    infileinfo_data(iFile)%iTimebound(1) = infileinfo_data(iFile-1)%iTimebound(2) + 1 ! the last index from the perivous nc file + 1
-    infileinfo_data(iFile)%iTimebound(2) = infileinfo_data(iFile-1)%iTimebound(2) + nTime ! the last index from the perivous nc file + 1
+    inputfileinfo(iFile)%iTimebound(1) = inputfileinfo(iFile-1)%iTimebound(2) + 1 ! the last index from the perivous nc file + 1
+    inputfileinfo(iFile)%iTimebound(2) = inputfileinfo(iFile-1)%iTimebound(2) + nTime ! the last index from the perivous nc file + 1
    endif
 
+   print*, iFile
+   print*, inputfileinfo(iFile)%iTimebound(1)
+   print*, inputfileinfo(iFile)%iTimebound(2)
+   print*, inputfileinfo(iFile)%infilename
+
   enddo
 
   ! close ascii file
   close(unit=unit,iostat=ierr); if(ierr/=0)then;message=trim(message)//'problem closing forcing file list'; return; end if
 
-  ! passing the first nc file as global file name to read
-  fname_qsim = trim(infileinfo_data(1)%infilename)
-
  END SUBROUTINE inFile_pop
 
 
+ ! *********************************************************************
+ ! private subroutine: get the two infiledata and convert the iTimebound of
+ ! the input_info_wm to match the input_info
+ ! *********************************************************************
+ SUBROUTINE inFile_corr_time(inputfileinfo,      & ! input: the structure of simulated runoff, evapo and
```

Comment:
The init_infile_pop populated both iTimebound(:) for runoff netcdf and abstraction/injection.
The second file might have different starting and ending point in time (the first one might start earlier, later or the same might end earlier or later). To have a reference to the iTime and iTimelocal, I decided to correct this before going into the model that will make infile_name simpler.
Examples:
1)if the first of second files start earlier than the runoff files its initial iTimebound will be negative (like -10000 days meaning the file is starting 30 years earlier)
2)if the first of second files start later than runoff then the iTimebound will be positive (like it start from 1000 day after the runoff file)
in the first example the model should not have an issue the iTime_local_wm start from 10000 while iTime and iTime_local are actually 1. in the second example the iTime_local_wm will be -1000 if the model start from the first iTime (from runoff time step) and the model stops. however if the starting point is set to sometime after 1000 days then the model works as it can read the second file values...
alternatively we can add this step into infile_name is that will be cleaner...
what is your suggestions on that?